android/app/src/main/java/com/mattermost/rnbeta/CustomPushNotification.java

```diff
@@ -173,19 +179,49 @@ protected void postNotification(int id, Notification notification) {
             style.setBigContentTitle(title);
             notification.setStyle(style)
                     .setContentTitle(summaryTitle);
-//                    .setNumber(numMessages);
         }
 
         // Let's add a delete intent when the notification is dismissed
-        Intent delIntent = new Intent(mContext, NotificationDismissReceiver.class);
+        Intent delIntent = new Intent(mContext, NotificationDismissService.class);
         delIntent.putExtra(NOTIFICATION_ID, notificationId);
-        PendingIntent deleteIntent = PendingIntent.getBroadcast(mContext, 0, delIntent, 0);
+        PendingIntent deleteIntent = NotificationIntentAdapter.createPendingNotificationIntent(mContext, delIntent, mNotificationProps);
         notification.setDeleteIntent(deleteIntent);
 
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
             notification.setGroup(GROUP_KEY_MESSAGES);
         }
 
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.N) {
+            try {
+                //Use reflection to remove all old actions
+                Field f = notification.getClass().getDeclaredField("mActions");
```

Comment:
What is this doing? Using reflection like this seems ridiculously hacky

Reply:
so there is no way to access `mActions`without reflection for the NotificationBuilder and we need that in order to clear it and set the "updated" action

Reply:
This action https://github.com/mattermost/mattermost-mobile/pull/871/files#diff-d29c058681ba74acb23997331d3d778eR222.

But I found a way around this and a bug that I need to fix